### PR TITLE
Add timestamptz operators

### LINKED ISSUE #316: Bug: Missing doc for arithmetics on date/timestamp/time/interval
### Where's the bug?

https://www.risingwave.dev/docs/latest/sql-function-datetime/

### Describe the bug

Everything listed in the following issue has been implemented except for `timestamp with time zone`
https://github.com/risingwavelabs/risingwave/issues/924

For reference this is the PostgreSQL document:
https://www.postgresql.org/docs/current/functions-datetime.html#OPERATORS-DATETIME-TABLE

### PR DESCRIPTION
**Pull request by bernscode**
Add timestamptz operators.

## Info
- **Description**: 
Add timestamptz operators.

- **Related code PR**: 
No related PR. This was a reported bug.

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/316

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Comment by xiangjinwu** [comment]
* We need to warn that interval can only contain hour/min/second but not year/month/day for timestamptz operators.
* This does not resolve #316 completely. I have listed the other missing operators in that issue.
* `The 'Z' at the end of timestamp with time zone stands for UTC time zone.` is a little bit repetitive. Maybe it belongs to the description of the type on [this page](https://risingwave.dev/docs/latest/sql-data-types/).

**Comment by bernscode** [comment]
@xiangjinwu Thanks for the feedback. I've updated and added the new operators to the latest commit.

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 1]
```diff
@@ -6,23 +6,50 @@ title: Date and time functions and operators
...
-| interval = interval → bool | Compares interval equality. | `interval '1' month = interval '30' day` → `t` |
-| real * interval → time | Multiplies an interval by a float. | `real '6.1' * interval '1' second` → `00:00:06.1` <br /> `interval '1' second * real '6.1'` → `00:00:06.1` |
+
+
+
+## Timestamp with time zone operators
+
+| Operation | Description | Example |
+| ----------- | ----------- | ----------- |
+| timestamptz + *interval_fixed* → timestamptz | Adds a fixed interval to a timestamp with time zone. See note below. | `'2022-03-13 01:00:00'::timestamp + interval '24' hour` → `2022-03-14 01:00:00` |
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-datetime.md]:
```suggestion
| timestamptz + *interval_fixed* → timestamptz | Adds a fixed interval to a timestamp with time zone. See note below. | `'2022-03-13 01:00:00Z'::timestamp with time zone + interval '24' hour` → `2022-03-14 01:00:00+00:00` |
```

**Review by xiangjinwu — COMMENTED** [review]
rest lgtm. Thank you

**Review by CharlieSYH — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-datetime.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: @xiangjinwu Excellent! Thanks for catching this! I've made the update to the latest commit.

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 2]
```diff
@@ -6,23 +6,50 @@ title: Date and time functions and operators
...
-| interval = interval → bool | Compares interval equality. | `interval '1' month = interval '30' day` → `t` |
-| real * interval → time | Multiplies an interval by a float. | `real '6.1' * interval '1' second` → `00:00:06.1` <br /> `interval '1' second * real '6.1'` → `00:00:06.1` |
+
+
+
+## Timestamp with time zone operators
+
+| Operation | Description | Example |
+| ----------- | ----------- | ----------- |
+| timestamptz + *interval_fixed* → timestamptz | Adds a fixed interval to a timestamp with time zone. See note below. | `''2022-03-13 01:00:00Z'::timestamp with time zone + interval '24' hour` → `2022-03-14 01:00:00+00:00` |
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-datetime.md]:
```suggestion
| timestamptz + *interval_fixed* → timestamptz | Adds a fixed interval to a timestamp with time zone. See note below. | `'2022-03-13 01:00:00Z'::timestamp with time zone + interval '24' hour` → `2022-03-14 01:00:00+00:00` |
```

**Review by xiangjinwu — APPROVED** [review]

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**